src/iidm/util/SV.cpp

```diff
@@ -76,114 +76,218 @@ double SV::getQ() const {
-    std::complex<double> z (r, x);
-    std::complex<double> y1(g1, b1);
-    std::complex<double> y2(g2, b2);
-    std::complex<double> u2 = computeU2(y1, y2, z, rho);
-    std::complex<double> s2 = computeS2(y1, y2, z, rho);
-    return {-s2.real(), -s2.imag(), std::abs(u2), std::arg(u2) * stdcxx::toDegrees};
+SV SV::otherSide(const TwoWindingsTransformer& twt) const {
+    return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), 0.0, 0.0, getRho(twt), getAlpha(twt));
 }
 
-SV SV::otherSide(const TwoWindingsTransformer& twt) const {
-    return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), getRho(twt));
+SV SV::otherSide(const TwoWindingsTransformer& twt, bool splitShuntAdmittance) const {
+    if (splitShuntAdmittance) {
+        return otherSide(getR(twt), getX(twt), getG(twt) * 0.5, getB(twt) * 0.5, getG(twt) * 0.5, getB(twt) * 0.5, getRho(twt), getAlpha(twt));
+    }
+    return otherSide(twt);
 }
 
 SV SV::otherSide(const Line& line) const {
-    return otherSide(line.getR(), line.getX(), line.getG1() + line.getG2(), line.getB1() + line.getB2(), 1);
+    return otherSide(line.getR(), line.getX(), line.getG1(), line.getB1(), line.getG2(), line.getB2(), 1.0, 0.0);
+}
+
+SV SV::otherSide(const DanglingLine& dl) const {
+    return otherSide(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
 }
 
-SV SV::otherSide(const DanglingLine& danglingLine) const {
-    return otherSide(danglingLine.getR(), danglingLine.getX(), danglingLine.getG() / 2.0, danglingLine.getB() / 2.0, danglingLine.getG() / 2.0, danglingLine.getB() / 2.0, 1);
+SV SV::otherSide(const DanglingLine& dl, bool splitShuntAdmittance) const {
+    if (splitShuntAdmittance) {
+        return otherSide(dl.getR(), dl.getX(), dl.getG() * 0.5, dl.getB() * 0.5, dl.getG() * 0.5, dl.getB() * 0.5, 1.0, 0.0);
+    }
+    return otherSide(dl);
 }
 
-double SV::otherSideA(double r, double x, double g1, double b1, double rho) const {
-    std::complex<double> z(r, x);
-    std::complex<double> y1(g1, b1);
-    std::complex<double> u2 = computeU2(y1, std::complex<double>(), z, rho);
-    return std::arg(u2) * stdcxx::toDegrees;
+SV SV::otherSide(const LinkData::BranchAdmittanceMatrix& adm) const {
+    std::complex<double> v;
+    std::complex<double> s;
+    Branch::Side otherSide;
+    if (m_side == Branch::Side::ONE) {
+        std::complex<double> v1 = std::polar(m_u, m_a * stdcxx::toRadians);
+        std::complex<double> s1(m_p, m_q);
+        v = voltageAtEnd2(adm, v1, s1);
+        s = flowAtEnd2(adm, v1, v);
+        otherSide = Branch::Side::TWO;
+    } else {
+        std::complex<double> v2 = std::polar(m_u, m_a * stdcxx::toRadians);
+        std::complex<double> s2(m_p, m_q);
+        v = voltageAtEnd1(adm, v2, s2);
+        s = flowAtEnd1(adm, v, v2);
+        otherSide = Branch::Side::ONE;
+    }
+    return {std::real(s), std::imag(s), std::abs(v), std::arg(v) * stdcxx::toDegrees, otherSide};
```

Comment:
I'm not a really big fan of this syntax :)
If you decide to do so, why don't you use it when you pass a complex at line 120 or just below at line 172?

Reply:
I'm not a fan either... but it was recommended by clang-tidy : 
```
/mnt/e/Projet_SRO/rte/powsybl-iidm4cpp-master/src/iidm/util/SV.cpp:168:12: error: avoid repeating the return type from the declaration; use a braced initializer list instead [modernize-return-braced-init-list,-warnings-as-errors]
    return SV(std::real(s), std::imag(s), std::abs(v), std::arg(v) * stdcxx::toDegrees, otherSide);
```

It is not problematic on line 120 or 172, maybe because they are constexpr?